UnityProject/Assets/Scripts/HealthV2/Living/Damage/Trauma/BodyPartTrauma.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+namespace HealthV2
+{
+	public class BodyPartTrauma : BodyPartFunctionality
+	{
+		[SerializeField] private List<TraumaLogic> traumaTypesOnBodyPart;
+		public List<TraumaLogic> TraumaTypesOnBodyPart => traumaTypesOnBodyPart;
+
+		private void Start()
+		{
+			if (RelatedPart == null)
+			{
+				Logger.LogError($"No component found on parent. Make sure to put this component on a child of the bodyPart");
+				return;
+			}
+
+			RelatedPart.OnDamageTaken += OnDamageTaken;
+		}
+
+		private void OnDestroy()
+		{
+			RelatedPart.OnDamageTaken -= OnDamageTaken;
+		}
+
+		public override void AddedToBody(LivingHealthMasterBase livingHealth)
+		{
+			var creatureTraumaAPI = livingHealth.GetComponent<CreatureTraumaManager>();
+			if (creatureTraumaAPI == null)
+			{
+				Logger.LogWarning($"[BodyPartTrauma/OnAddBodyPart] - No high level trauma manager detected on creature." +
+				                  $"Functionalities like trauma healing may not be available for this body part.");
+				return;
+			}
+			creatureTraumaAPI.Traumas.Add(RelatedPart, this);
+		}
+
+		public override void RemovedFromBody(LivingHealthMasterBase livingHealth)
+		{
+			var creatureTraumaAPI = livingHealth.GetComponent<CreatureTraumaManager>();
+			if (creatureTraumaAPI == null) return;
+			creatureTraumaAPI.Traumas.Remove(RelatedPart);
+		}
+
+		private void OnDamageTaken(BodyPartDamageData data)
```

Comment:
Might be worth putting this inside of  body part functionality , So other stuff can plug into it easily